presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -635,6 +642,28 @@ public void setDriverInstanceCount(int driverInstanceCount)
             }
             this.driverInstanceCount = OptionalInt.of(driverInstanceCount);
         }
+
+        public void intersectDynamicFilter(TupleDomain<DynamicFilters.Descriptor> newPredicate)
+        {
+            if (newPredicate.isAll()) {
+                return;
+            }
+            synchronized (dynamicFilterLock) {
+                dynamicFilter = dynamicFilter.intersect(newPredicate);
+            }
+        }
+
+        public TupleDomain<ColumnHandle> getDynamicFilter(TableScanNode tableScan)
+        {
+            Map<Symbol, ColumnHandle> assignments = tableScan.getAssignments();
+            synchronized (dynamicFilterLock) {
+                // Skips symbols irrelevant to this table scan node.
+                return dynamicFilter
+                        .transform(DynamicFilters.Descriptor::getInput)
+                        .transform(Symbol::from)
+                        .transform(assignments::get);
```

Comment:
`assignments::get` could return null potentially

Reply:
IIUC, it should be fine since `TupleDomain#transform` should skip the symbols that doesn't have a corresponding column in `assignments`:
https://github.com/prestosql/presto/blob/bfac56d6b34ec488a3ff132055fc0aca25d5448e/presto-spi/src/main/java/io/prestosql/spi/predicate/TupleDomain.java#L361-L383